examples/ComplexExample/app/components/card/detail.js

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import * as Actions from '../../actions/detail'
+import {
+    View,
+    Text,
+    StyleSheet,
+    Button,
+    ScrollView,
+    TouchableOpacity
+} from 'react-native'
+import Icon from 'react-native-vector-icons/MaterialIcons'
+import Styles from '../../styles/card.style'
+import { headerStyle, titleStyle } from '../../styles/header.style'
+import ButtonRight from '../header/button-right'
+import ButtonBack from '../header/button-back'
+
+function mapStateToProps(state) {
+    return {
+        detail: state.detail
+    }
+}
+
+class Detail extends Component {
+
+    static navigationOptions = {
+        title: ({ state }) => `${state.params.item.title}`,
+        header: ({ state, goBack }, defaultHeader) => ({
+            style: headerStyle,
+            titleStyle: titleStyle,
+            right: (state.params && state.params.right),
+            left: (<ButtonBack goBack={goBack} />)
+        })
+    }
+
+    // If no detail items loaded, load now
+    componentWillMount() {
+
+        const { detail, navigation } = this.props
+        const { index } = navigation.state.params.item
+
+        if (!detail[index])
+            this.props.fetchDetailState({ limit: 10 })
+    }
+
+    // Link right header button to component
+    // In this case we want the button to bookmark the detail item
+    componentDidMount() {
+
+        const { navigation } = this.props
+        const { index } = navigation.state.params.item
+
+        const params = {
+            right: (<ButtonRight
```

Comment:
Thats a shame. I didn't find any other way to call methods in context (this.bookmark in this case) from the navigation buttons due to the static navigationOptions. Will move to componentWillMount for the time being.